kahuna/public/js/search-query/service.js

```diff
@@ -0,0 +1,32 @@
+import angular from 'angular';
+import Rx from 'rx';
+import '../util/rx';
+
+export const searchQueryService = angular.module('gr.searchQuery.service', ['util.rx']);
+
+searchQueryService.factory('searchQueryService', [function() {
+    const query$ = new Rx.Subject();
+
+    // TODO: When we update rx, we need to swap the seed / acc around
+    const q$ = query$.scan('', (state, fn) => fn(state).trim());
+
+    function hasLabel(q, label) {
+        // TODO: It'd be nice to build this up from an API
+        const r = new RegExp(`[#|label\:]${label}\\b`);
```

Comment:
No, you're right it doesn't.
[Updated](https://github.com/guardian/grid/commit/d84af857546d103811715310a0abf993407e1cb3)